materialhelptutorial/src/main/java/za/co/riggaroo/materialhelptutorial/MaterialTutorialFragment.java

```diff
@@ -65,9 +66,12 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
         if (tutorialItem.getBackgroundImageRes() != -1) {
             Glide.with(this).load(tutorialItem.getBackgroundImageRes()).into(imageViewBack);
         }
-        if (tutorialItem.getForegroundImageRes() != -1) {
+        if (tutorialItem.getForegroundImageRes() != -1 && !tutorialItem.isGif()) {
             Glide.with(this).load(tutorialItem.getForegroundImageRes()).into(imageViewFront);
         }
+        if (tutorialItem.getForegroundImageRes() != -1 && tutorialItem.isGif()){
+            Glide.with(this).load(tutorialItem.getForegroundImageRes()).asGif().diskCacheStrategy(DiskCacheStrategy.SOURCE).into(imageViewFront);
```

Comment:
You don't need this distinction because you're loading from resources. This will just copy the already big GIF files to the cache which won't be faster because the APK and the cache are both on device / sd card. This just wastes space for users. Also `asGif()` is not needed because Glide knows if it's still or not.